src/libPMacc/include/particles/ParticlesBase.kernel

```diff
@@ -49,181 +52,301 @@ getPreviousFrameAndRemoveLastFrame( const typename T_ParticleBox::FramePtr& fram
     return result;
 }
 
+template< uint32_t T_worker >
 struct KernelShiftParticles
 {
     /*! This kernel move particles to the next supercell
      * This kernel can only run with a double checker board
      */
-    template<class T_ParBox, class Mapping>
-    DINLINE void operator()( T_ParBox pb, Mapping mapper ) const
+    template<
+        class T_ParBox,
+        class Mapping
+    >
+    DINLINE void operator()(
+        T_ParBox pb,
+        Mapping mapper
+    ) const
     {
-        typedef T_ParBox ParBox;
-        typedef typename ParBox::FrameType FrameType;
-        typedef typename ParBox::FramePtr FramePtr;
+        using ParBox = T_ParBox;
```

Comment:
Yes, but this is a full rewrite the new helper are hard to use with the old syntax thus I rewrite everything with the new code style.
All other kernel in this file will also be refactored but it is easier to give the changes step by step.